sklearn/semi_supervised/tests/test_label_propagation.py

```diff
@@ -73,18 +78,19 @@ def test_label_spreading_closed_form():
     X, y = make_classification(n_classes=n_classes, n_samples=200,
                                random_state=0)
     y[::3] = -1
-    clf = label_propagation.LabelSpreading().fit(X, y)
-    # adopting notation from Zhou et al (2004):
-    S = clf._build_graph()
-    Y = np.zeros((len(y), n_classes + 1))
-    Y[np.arange(len(y)), y] = 1
-    Y = Y[:, :-1]
-    for alpha in [0.1, 0.3, 0.5, 0.7, 0.9]:
-        expected = np.dot(np.linalg.inv(np.eye(len(S)) - alpha * S), Y)
-        expected /= expected.sum(axis=1)[:, np.newaxis]
-        clf = label_propagation.LabelSpreading(max_iter=10000, alpha=alpha)
-        clf.fit(X, y)
-        assert_array_almost_equal(expected, clf.label_distributions_, 4)
+    for sparse_or_dense in SPARSE_OR_DENSE:
```

Comment:
could you please parameterize the test for this one and `alpha`? Also for the other tests

Reply:
@adrinjalali: I didn't write the original tests, so I'm probably not the right one to ask for improving these with regard to parametrization.